lib/components/mp4muxer/helpers/boxbuilder.js

```diff
@@ -216,6 +216,17 @@ class BoxBuilder {
     box.add('data', data)
     return box
   }
+
+  setPresentationTime (trackId, ntpTimestamp) {
+    // Before updating the baseMediaDecodeTime, we check if
+    // there is already a base NTP time to use as a reference
+    // for computing presentation times.
+    if (!this.ntpPresentationTime && ntpTimestamp && trackId === this.videoTrackId) {
+      const trackOffset = trackId - 1
+      const trackData = this.trackData[trackOffset]
+      this.ntpPresentationTime = ntpTimestamp - 1000 * (trackData.baseMediaDecodeTime / trackData.clockrate)
```

Comment:
`this.ntpPresentationTime` should be initialized to `0` in the moov section (so that it is properly reset when a new movie is started). You can do this after e.g. `this.sequenceNumber = 0` (~line 111)